src/sinks/util/mod.rs

```diff
@@ -104,14 +106,315 @@ impl<I> EncodedEvent<I> {
+                    "KERN" => 0,
+                    "USER" => 1,
+                    "MAIL" => 2,
+                    "DAEMON" => 3,
+                    "AUTH" => 4,
+                    "SYSLOG" => 5,
+                    "LPR" => 6,
+                    "NEWS" => 7,
+                    "UUCP" => 8,
+                    "CRON" => 9,
+                    "AUTHPRIV" => 10,
+                    "FTP" => 11,
+                    "NTP" => 12,
+                    "SECURITY" => 13,
+                    "CONSOLE" => 14,
+                    "SOLARIS-CRON" => 15,
+                    "LOCAL0" => 16,
+                    "LOCAL1" => 17,
+                    "LOCAL2" => 18,
+                    "LOCAL3" => 19,
+                    "LOCAL4" => 20,
+                    "LOCAL5" => 21,
+                    "LOCAL6" => 22,
+                    "LOCAL7" => 23,
+                    _ => 24,
+                };
+                if num == 24 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown facility"));
+                } else {
+                    return Ok(Facility::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
+fn default_severity() -> Severity {
+    Severity::Fixed(0)
+}
+
+fn deserialize_severity<'de, D>(d: D) -> Result<Severity, D::Error>
+    where D: de::Deserializer<'de>
+{
+    let value: String = String::deserialize(d)?;
+    let num_value = value.parse::<u8>();
+    match num_value {
+        Ok(num) => {
+            if num > 6 {
+                return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"severity number too large"))
+            } else {
+                return Ok(Severity::Fixed(num))
+            }
+        }
+        Err(_) => {
+            if let Some(field_name) = value.strip_prefix("$.message.") {
+                return Ok(Severity::Field(field_name.to_string()));
+            } else {
+                let num = match value.to_uppercase().as_str() {
+                    "EMERGENCY" => 0,
+                    "ALERT" => 1,
+                    "CRITICAL" => 2,
+                    "ERROR" => 3,
+                    "WARNING" => 4,
+                    "INFORMATIONAL" => 5,
+                    "DEBUG" => 6,
+                    _ => 7,
+                };
+                if num > 6 {
+                    return Err(de::Error::invalid_value(de::Unexpected::Unsigned(num as u64), &"unknown severity"))
+                } else {
+                    return Ok(Severity::Fixed(num))
+                }
+            }
+        }
+    }
+}
+
 /**
  * Enum representing different ways to encode events as they are sent into a Sink.
  */
-#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 #[serde(rename_all = "snake_case")]
 pub enum Encoding {
     Text,
     Json,
+    Syslog(SyslogConf),
+}
```

Comment:
@vimalk78 do you mean 'consider' or 'require'.  Does this enum represent the encodings which a sink "may" support or "must" support?

Reply:
since it is a part of enum, and if a `match` is done on the enum the enum MUST be matched on all variants.

Reply:
The compiler would've flagged a mismatch between an enum and an non-exhaustive match but it didn't because we don't compile the papertrail sink.
But please consider the larger context. This PR will not become an upstream submission. In the current version of the upstream encodings are handled differently (https://github.com/vectordotdev/vector/tree/master/lib/codecs/src/encoding/format) and the syslog code would live there.

Reply:
if this implementation is specifically for us, just to be safe, we should put a feature flag to isolate these changes
```
#[cfg(feature = "syslog-rh")]
```

Reply:
in more recent releases, (e.g. `master` which now is `0.27`) , this enum does not exist. What is there is more sophisticated way of specifying [EncodingWithFraming](https://github.com/vectordotdev/vector/blob/master/src/codecs/encoding/config.rs#L61-L67) , and encoding can be chosen from a much larger set of serializers in  [SerilizerConfig](https://github.com/vectordotdev/vector/blob/master/lib/codecs/src/encoding/mod.rs#L191-L264)

One of these is `LogFmt` , and [this code](https://github.com/vectordotdev/vector/blob/master/src/components/validation/resources/mod.rs#L138-L158) seems to be suggesting that LogFmt can be used as serializer for syslog. So perhaps in future releases, we can reuse what already exists to realize a SyslogEncoder.